doc/source/contribute.rst

```diff
@@ -1,49 +1,175 @@
 
+Before starting any development, you'll need to create an isolated larray
+development environment. One of the easiest way to do it is via `Anaconda` or `miniconda`
+(while this not mandatory):
 
-Installing the module
----------------------
+- Install either `Anaconda <https://www.anaconda.com/download/>`_ or `miniconda
+  <https://conda.io/miniconda.html>`_ as :ref:`suggest earlier <contributing.tools>`
+- Make sure your conda is up to date (``conda update conda``)
+- Make sure that you have :ref:`cloned the repository <contributing.getting_code>`
+- ``cd`` to the *larray* source directory
 
-You could install LArray in the standard way: ::
+We'll now kick off a two-step process:
 
-  > python setup.py install
+1. Install the build dependencies
+
+.. code-block:: none
+
+   # Create and activate the build environment
+   conda create -n larray_dev numpy pandas pytables pyqt qtpy matplotlib xlrd openpyxl xlsxwriter pytest
+   conda activate larray_dev
+
+This will create the new environment, and not touch any of your existing environments,
+nor any existing Python installation.
+
+To view your environments::
+
+      conda info -e
+
+To return to your root environment::
+
+      conda deactivate
+
+See the full conda docs `here <http://conda.pydata.org/docs>`_.
+
+2. Build and install larray
+
+You could install LArray in the standard way:
+
+.. code-block:: none
+
+  python setup.py install
 
 but in that case you need to "install" it again every time you change it. When developing, it is usually more
-convenient to use: ::
+convenient to use:
+
+.. code-block:: none
+
+  python setup.py develop
+
+This creates some kind of symbolic link between your python installation "modules"
+directory and your repository, so that any change in your local copy is automatically
+usable by other modules.
+
+At this point you should be able to import larray from your locally built version::
+
+   $ python  # start an interpreter
+   >>> import larray
+   >>> larray.__version__
+   '0.29-dev'
 
-  > python setup.py develop
 
-This creates some kind of symlink between your python installation "modules" directory and your repository, so that any
-change in your local copy is automatically usable by other modules.
+Starting to contribute
+----------------------
 
+For developing the LArray library, we follow the `Forking Workflow
+<https://gist.github.com/Chaser324/ce0505fbed06b947d962>`_.
+In the :ref:`Getting code <contributing.getting_code>` section,
+we have already explained how to get a fork of the main larray repository.
 
-Updating your local copy with remote changes
---------------------------------------------
+To make a contribution, please follow the steps described bellow.
 
-::
+Step 1: Create a new branch
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-  > git pull  # or git fetch + git merge
+You want your master branch to reflect only production-ready code, so create a
+feature branch for making your changes. For example::
 
+    git checkout -b issue-to-fix
```

Comment:
s/issue-to-fix/issue123/ ?

Also, you sometimes use issue-to-fix and sometimes shiny-new-feature. Pick one of the (now) three possibilities and stick to it.